diff --git a/src/Vulkan/GraphicsPipeline.cpp b/src/Vulkan/GraphicsPipeline.cpp
index 7fa4a79..dc8d527 100644
--- a/src/Vulkan/GraphicsPipeline.cpp
+++ b/src/Vulkan/GraphicsPipeline.cpp
@@ -120,7 +120,10 @@ GraphicsPipeline::setColorBlendStates(std::span<const VkPipelineColorBlendAttach
 GraphicsPipeline& GraphicsPipeline::setColorBlendOptions(VkPipelineColorBlendStateCreateFlags flags,
                                                          const rstd::array<float, 4>& constants) {
     m_color.flags = flags;
-    std::copy(constants.begin(), constants.end(), std::begin(m_color.blendConstants));
+    m_color.blendConstants[0] = constants.at(usize(0));
+    m_color.blendConstants[1] = constants.at(usize(1));
+    m_color.blendConstants[2] = constants.at(usize(2));
+    m_color.blendConstants[3] = constants.at(usize(3));
     return *this;
 }
 
