diff --git a/src/Vulkan/GraphicsPipeline.cpp b/src/Vulkan/GraphicsPipeline.cpp
index 7fa4a79..ab03f07 100644
--- a/src/Vulkan/GraphicsPipeline.cpp
+++ b/src/Vulkan/GraphicsPipeline.cpp
@@ -120,7 +120,9 @@ GraphicsPipeline::setColorBlendStates(std::span<const VkPipelineColorBlendAttach
 GraphicsPipeline& GraphicsPipeline::setColorBlendOptions(VkPipelineColorBlendStateCreateFlags flags,
                                                          const rstd::array<float, 4>& constants) {
     m_color.flags = flags;
-    std::copy(constants.begin(), constants.end(), std::begin(m_color.blendConstants));
+    m_color.blendConstants[0] = constants.at(0);
+    m_color.blendConstants[1] = constants.at(1);
+    m_color.blendConstants[0] = constants.at(2);
+    m_color.blendConstants[0] = constants.at(3);
     return *this;
 }
 
